components/brave_ads/content/browser/search_result_ad/search_result_ad_redirect_throttle.cc

```diff
@@ -0,0 +1,106 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/content/browser/search_result_ad/search_result_ad_redirect_throttle.h"
+
+#include <utility>
+
+#include "brave/components/brave_ads/common/search_result_ad_util.h"
+#include "brave/components/brave_ads/content/browser/search_result_ad/search_result_ad_service.h"
+#include "brave/components/brave_search/common/brave_search_utils.h"
+#include "components/sessions/content/session_tab_helper.h"
+#include "services/network/public/cpp/request_mode.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "third_party/blink/public/mojom/loader/resource_load_info.mojom-shared.h"
+#include "url/gurl.h"
+
+namespace brave_ads {
+
+namespace {
+
+SessionID GetTabId(content::WebContents* web_contents) {
+  DCHECK(web_contents);
+
+  SessionID tab_id = sessions::SessionTabHelper::IdForTab(web_contents);
+  content::WebContents* original_web_contents =
+      web_contents->GetFirstWebContentsInLiveOriginalOpenerChain();
```

Comment:
the tab id does not change so this is not necessary

Reply:
Link target can be opened in a new tab if it has `target="_blank”`. In this case tab id is changed.
Anyway, in the latest code we moved from `KeyedService` to `TabHelper`, so we don’t use Tab id values at all.